sys/kern/vm_physmem.c

```diff
@@ -77,20 +67,13 @@ static void *vm_boot_alloc(size_t n) {
 
   assert(seg != NULL);
 
-  for (void *va = align(begin, PAGESIZE); va < end; va += PAGESIZE) {
-    paddr_t pa = seg->start;
-    seg->start += PAGESIZE;
-    if (--seg->npages == 0) {
-      TAILQ_REMOVE(&seglist, seg, seglink);
-      seg = TAILQ_FIRST(&seglist);
-    }
-
-    pmap_kenter((vaddr_t)va, pa, VM_PROT_READ | VM_PROT_WRITE, 0);
-  }
+  void *va = phys_to_dmap(seg->start);
 
-  vm_kernel_end += n;
+  seg->start += n;
+  seg->npages -= n / PAGESIZE;
+  assert(seg->npages);
```

Comment:
Could you explain this assertion?

Reply:
This assert is needles as the condition is ensured by the condition of the preceding `while` loop. I removed it.